### PR TITLE
OPENDINGUX: Add support for Opendingux Beta

### PR DESCRIPTION
**Pull request by citral23**
Hello, this is a V2 for my upstreaming proposal of Opendingux beta support.

It now uses an improved, unified build script, that takes the cpu variant as parameter (cf backends/platform/sdl/opendingux/README.BUILD).

It uses the keymapper API instead of the old custom, hardcoded events backend.

Thorough testing has ensure performance is adequate (for instance, discworld runs ok on the weakest CPU) and all options that have proven too demanding removed.

Hugetlb support has been dropped as it hasn't proven to be production worthy, and compilation flags refined.

I'm of course open to questions/remarks here or on discord.

KR

### CODE REVIEW

**Comment by sev-** [comment]
Before even looking closely at the changes, could you please bring the commit log messages in line with our Commit Guidelines? https://wiki.scummvm.org/index.php?title=Commit_Guidelines

File: backends/platform/sdl/opendingux/build_odbeta.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+CONFIG="./configure --host=opendingux --enable-release --disable-detection-full"
```
[the remark this answers is not in the document]
Comment (sev-) [reply backends/platform/sdl/opendingux/build_odbeta.sh:3]: It is fine and acceptable to have platform-specific build scripts like this for the release reproducibility.

File: backends/platform/sdl/opendingux/build_odbeta.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+if [ $target == "gcw0" ]; then
+	SYSROOT=$TOOLCHAIN/mipsel-$target2-linux-uclibc
+else
+	SYSROOT=$TOOLCHAIN/mipsel-$target2-linux-musl
+fi
+
+export PATH=$TOOLCHAIN/usr/bin:$SYSROOT/usr/include:$TOOLCHAIN/bin:$PATH
+export CXX=mipsel-linux-g++
+export CXXFLAGS="-funsigned-char" # workaround for a scummvm tolower() bug when adding games
```
[the remark this answers is not in the document]
Comment (sev-) [reply backends/platform/sdl/opendingux/build_odbeta.sh:39]: This is then a bug that has to be fixed and we need to use `byte` where it is needed. Could you please be a bit more specific regarding this "tolower() bug"?

[the remark this answers is not in the document]
Comment (lephilousophe) [reply backends/platform/sdl/opendingux/build_odbeta.sh:39]: What I don't understand with this is that all common platforms we use are considering `char` as `signed char` so MIPS is not special on this.

[the remark this answers is not in the document]
Comment (sev-) [reply backends/platform/sdl/opendingux/build_odbeta.sh:39]: Amiga has problems with signed char every now and then.

File: backends/platform/sdl/opendingux/build_odbeta.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (lephilousophe) [reply backends/platform/sdl/opendingux/build_odbeta.sh:3]: In fact what bothers me with this script is that it completely mangles what configure script does but outside it.
For example the DEFINES variable will be added to defines in configure.

Usual workflow when not building for release is to run `configure --host=target` and the configure script will setup most of settings according to the host.
I think about highres, hq-scalers, 16bit, detection-full and the `-DLEPUS`, `-DRS90 -DDISABLE_FANCY_THEMES` flags.
Without all of these settings, the configure script alone is useless.

On the other hand, I agree that the toolchain setup and make commands shouldn't be in the configure script.

**Comment by sev-** [comment]
@citral23 Thank you for your work on this. Would you like to join the team as the official porter? If so, please ping me on Discord and I'll explain the procedure.

**Pull request merged**